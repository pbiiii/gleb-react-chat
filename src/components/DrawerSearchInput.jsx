import React from 'react';
import {withStyles} from '@material-ui/core/styles';
import Input from '@material-ui/core/Input';

const styles = theme => ({
    toolbar: {
        ...theme.mixins.toolbar,
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
    },
    toolbarInput: {
        width: '90%',
    },
});

const DrawerSearchInputComponent = ({classes}) => {
    return (
        <div className={classes.toolbar}>
            <Input
                className={classes.toolbarInput}
                placeholder={'Search chats...'}
            />
        </div>
    );
}

export const DrawerSearchInput = withStyles(styles)(DrawerSearchInputComponent);