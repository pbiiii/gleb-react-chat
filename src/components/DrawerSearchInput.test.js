/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import DrawerSearchInput from './DrawerSearchInput';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<DrawerSearchInput />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('render correctly', () => {
    const tree = renderer
        .create(<DrawerSearchInput />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
