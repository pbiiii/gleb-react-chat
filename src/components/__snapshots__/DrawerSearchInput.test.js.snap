// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="DrawerSearchInput-toolbar-33"
>
  <div
    className="MuiInputBase-root-48 MuiInput-root-35 MuiInput-underline-39 DrawerSearchInput-toolbarInput-34"
    onClick={[Function]}
  >
    <input
      className="MuiInputBase-input-58 MuiInput-input-43"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Search chats..."
      type="text"
    />
  </div>
</div>
`;
